app/models/prisoner_validation.rb

```diff
@@ -17,10 +26,30 @@ def active_offender_exists
 
     unless offender.valid?
       errors.add :base, PRISONER_NOT_EXIST
+      return
+    end
+
+    unless prisoner_location.api_call_successful?
```

Comment:
I think here we need to check for the offender being valid before calling the prisoner_location